crates/katana/primitives/src/genesis/json.rs

```diff
@@ -265,81 +289,97 @@ impl TryFrom<GenesisJson> for Genesis {
     type Error = GenesisJsonError;
 
     fn try_from(value: GenesisJson) -> Result<Self, Self::Error> {
-        let mut classes: HashMap<ClassHash, GenesisClass> = value
-            .classes
-            .into_par_iter()
-            .map(|entry| {
-                let GenesisClassJson { class, class_hash } = entry;
-
-                let artifact = match class {
-                    PathOrFullArtifact::Artifact(artifact) => artifact,
-                    PathOrFullArtifact::Path(path) => {
-                        return Err(GenesisJsonError::UnresolvedClassPath(path));
-                    }
-                };
-
-                let sierra = serde_json::from_value::<SierraClass>(artifact.clone());
-
-                let (class_hash, compiled_class_hash, sierra, casm) = match sierra {
-                    Ok(sierra) => {
-                        let casm: ContractClass = serde_json::from_value(artifact)?;
-                        let casm = CasmContractClass::from_contract_class(casm, true)?;
-
-                        // check if the class hash is provided, otherwise compute it from the
-                        // artifacts
-                        let class_hash = class_hash.unwrap_or(sierra.class_hash()?);
-                        let compiled_hash = casm.compiled_class_hash().to_be_bytes();
-
-                        (
-                            class_hash,
-                            FieldElement::from_bytes_be(&compiled_hash)?,
-                            Some(Arc::new(sierra.flatten()?)),
-                            Arc::new(CompiledContractClass::V1(CompiledContractClassV1::try_from(
-                                casm,
-                            )?)),
-                        )
-                    }
-
-                    // if the artifact is not a sierra contract, we check if it's a legacy contract
-                    Err(_) => {
-                        let casm: CompiledContractClassV0 =
-                            serde_json::from_value(artifact.clone())?;
-
-                        let class_hash = if let Some(class_hash) = class_hash {
-                            class_hash
-                        } else {
-                            let casm: LegacyContractClass =
-                                serde_json::from_value(artifact.clone())?;
-                            casm.class_hash()?
+                let mut name_to_class_hash = HashMap::new();
+                let mut classes: HashMap<ClassHash, GenesisClass> = value
+                    .classes
+                    .into_par_iter()
+                    .map(|entry| {
+                        let GenesisClassJson { class, class_hash, name } = entry;
+                        // Use name if present
+                        if let (Some(name), Some(class_hash)) = (name, class_hash) {
+                    name_to_class_hash.insert(name, class_hash);
+                }
```

Comment:
```suggestion
```
I think this may be removed, in favor of the code at the end of this function. If the name is here, we insert the mapping. Because at the end of this function, a `class_hash` will always be defined (from the file, or from the artifacts).